apps/analog-app/src/app/pages/form-action.directive.ts

```diff
@@ -0,0 +1,40 @@
+import { Directive, inject, input, output } from '@angular/core';
+import { Router } from '@angular/router';
+
+export type ActionResult = {
+  type: string;
+};
+
+@Directive({
+  selector: 'form[action],form[method]',
+  host: {
+    '(submit)': `submitted($event)`,
+  },
+  standalone: true,
+})
+export class FormActionDirective {
+  action = input<string>('');
+  onSuccess = output<unknown>();
+  onError = output<unknown>();
+  state = output<'submitting' | 'error'>();
+  router = inject(Router);
+
+  submitted($event: { target: HTMLFormElement } & Event) {
+    $event.preventDefault();
+    const body = new FormData($event.target);
+
+    fetch(this.action() || `/api/_analog/pages${window.location.pathname}`, {
+      method: $event.target.method,
+      body,
+    }).then((res) => {
```

Comment:
question: I'm not against `fetch`, but would it be possible to support/use HttpClient? Does the internal implementation even matter? I genuinely don't know.

Reply:
Its possible to use HttpClient, but I don't think it matters. fetch is more flexible out of the box in terms of determining what to do with the response